tests/bootstrap_impls/venv_relative_path_tests.bzl

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Unit tests for relative_path computation"
+
+load("@rules_testing//lib:test_suite.bzl", "test_suite")
+load("//python/private:py_executable_bazel.bzl", "relative_path")  # buildifier: disable=bzl-visibility
+
+_tests = []
+
+def _relative_path_test(env):
+    # Basic test cases
+
+    env.expect.that_str(
+        relative_path(
+            from_ = "a/b",
+            to = "c/d",
+        ),
+    ).equals("../../c/d")
+
+    env.expect.that_str(
+        relative_path(
+            from_ = "../a/b",
+            to = "../c/d",
+        ),
+    ).equals("../../c/d")
+
+    env.expect.that_str(
+        relative_path(
+            from_ = "../a/b",
+            to = "../../c/d",
+        ),
+    ).equals("../../../c/d")
```

Comment:
Is this correct?
Lets say current dir is `/foo/bar/`
Then `../a/b` maps to `/foo/a/b`
Then `../../c/d` maps to `/foo/c/d`
Then the relative path of `/foo/c/d` with respect to `/foo/a/b` is `../../c/d`

Reply:
I don't think this was incorrect.

If current dir was `/foo/bar`, then `../../c/d` maps to `/c/d, not `/foo/c/d`

So from `/foo/a/b` to `/c/d` is indeed `../../../c/d`